src/parser/grammar.spec.ts

```diff
@@ -1,15 +1,16 @@
 import { expect } from 'chai';
-import { strictGrammars, tolerantGrammars } from '~/parser/grammar';
+import { placeholderGrammars, strictGrammars, tolerantGrammars } from '~/parser/grammar';
 
 describe('Unit: liquidHtmlGrammar', () => {
   const grammars = [
     { mode: 'strict', grammar: strictGrammars },
     { mode: 'tolerant', grammar: tolerantGrammars },
+    { mode: 'completion', grammar: placeholderGrammars },
   ];
 
-  grammars.forEach(({ mode, grammar }) => {
-    describe(`Case: ${mode}`, () => {
-      it('should parse or not parse HTML+Liquid', () => {
+  describe(`Case: common to all grammars`, () => {
+    it('should parse or not parse HTML+Liquid', () => {
+      grammars.forEach(({ grammar }) => {
```

Comment:
Its a small nit but we lose some test granularity with moving the grammar traversal into a single unit test.

Would it make more sense to use [describe each](https://jestjs.io/docs/api#describeeachtablename-fn-timeout) here instead?

Reply:
While this repo is using mocha and not vitest, the [vitest plugin has issues with code-generated describe/it statements](https://github.com/vitest-dev/vscode#testeach-is-not-working). I did this small move to make it easier for us to transition if we wanted to unify everything at some point.

I agree that the test failing wouldn't directly be evident from the test though, but I feel like that's should be an easy enough thing to find out if you're refactoring the ohm grammar based on the changes you do.